docs/rendering/modelextensions/facedata.md

```diff
@@ -0,0 +1,109 @@
+Face Data
+=========
+
+In a vanilla "elements" model, additional data about an element's faces can be specified at either the element level or the face level. Faces which do not specify their own face data will fall back to the element's face data or a default if no face data is specified at the element level.
+
+To use this extension for a generated item model, the model must be loaded through the `forge:item_layers` model loader due to the vanilla item model generator not being extended to read this additional data.
+
+All values of the face data are optional.
+
+Elements Model
+--------------
+
+In vanilla "elements" models the face data applies to the face it is specified in or all faces of the element it is specified in which don't have their own face data.
+
+The additional data can be specified in the two ways shown in this example:
+```js
+{
+  "elements": [
+    {
+      "forge_data": {
+        "color": "0xFFFF0000",
+          "block_light": 15,
+          "sky_light": 15,
+          "ambient_occlusion": false
+      },
+      "faces": {
+        "north": {
+          "forge_data": {
+            "color": "0xFFFF0000",
+            "block_light": 15,
+            "sky_light": 15,
+            "ambient_occlusion": false
+          },
+          // ...
+        },
+        // ...
+      },
+      // ...
+    }
+  ]
+}
+```
+
+Generated Item Model
+--------------------
+
+In generated item models loaded through the `forge:item_layers` loader the face data applies to all faces generated from the texture specified for the layer the respective entry is associated with.
+
+The additional data has to be specified at the top level with the entry names in the `forge_data` object being the layer index the respective entry belongs to (in the example layer 1 will be tinted red and glow at full brightness):
+```js
+{
+  "textures": {
+    "layer0": "minecraft:item/stick",
+    "layer1": "minecraft:item/glowstone_dust"
+  },
+  "forge_data": {
+    "1": {
+      "color": "0xFFFF0000",
+      "block_light": 15,
+      "sky_light": 15,
+      "ambient_occlusion": false
+    }
+  }
+}
+```
+
+Parameters
+----------
+
+### Color
+
+Specifying a color value with the `color` entry will apply that color as a tint to the quads. Defaults to `0xFFFFFFFF` (white, fully opaque). The color must be in the `ARGB` format packed into a 32-bit integer and can be specified as either a hexadecimal string (`"0xAARRGGBB"`) or as a decimal integer literal (JSON does not support hexadecimal integer literals).
+
+!!! warning
+    If the color is specified as a 24-bit integer (no matter the representation), the resulting color will be fully transparent due to the alpha component being 0.
+
+This can be used as a replacement for tinting with [`BlockColor` and `ItemColor`][tinting] if the color values are constant.
+
+### Block and Sky Light
```

Comment:
Skylight is usually spelled as the same word.

Reply:
Everything I can find in the game code writes it as two words. It would also look weird to have two different ways of spelling the two light types.

Reply:
Fair enough